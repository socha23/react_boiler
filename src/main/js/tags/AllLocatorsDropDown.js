import React from 'react'
import {connect} from 'react-redux'
import LocatorValue from './LocatorValue'
import Select from 'react-select'

const optionRenderer = (item) => <LocatorValue locator={item}/>;

const LocatorDropDown = ({id, value, items, onChange}) => <Select
    value={value}
    options={items}
    labelKey="name"
    valueKey="id"
    optionRenderer={optionRenderer}
    valueRenderer={optionRenderer}
    onChange={(i) => onChange(i ? i.id : null)}
    clearable={false}
    id={id}
/>;

const mapStateToProps = (state, ownProps) => ({
    items: state.locators.items
});

export default connect(mapStateToProps)(LocatorDropDown);
