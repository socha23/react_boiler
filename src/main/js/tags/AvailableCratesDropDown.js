import React from 'react'
import {connect} from 'react-redux'
import {availableCrates} from './locatorHelpers'
import LocatorValue from './LocatorValue'
import Select from 'react-select'

const optionRenderer = (item) => <LocatorValue locator={item}/>;

const CrateDropDown = ({id, value, items, onChange}) => <Select
    value={value}
    options={items}
    labelKey="name"
    valueKey="id"
    optionRenderer={optionRenderer}
    valueRenderer={optionRenderer}
    onChange={(i) => onChange(i ? i.id : null)}
    clearable={false}
    id={id}
/>;

const mapStateToProps = (state, ownProps) => ({
    items: availableCrates(state.artifacts.items, state.locators.items, ownProps.forArtifact)
});

export default connect(mapStateToProps)(CrateDropDown);
