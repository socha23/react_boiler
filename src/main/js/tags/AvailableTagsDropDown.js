import React from 'react'
import {connect} from 'react-redux'
import Select from 'react-select'

import {availableTags} from './tagHelpers'
import TagValue from './TagValue'

const optionRenderer = (item) => <TagValue tag={item} inline/>;

const TagDropDown = ({id, value, items, onChange}) => <Select
    value={value}
    options={items}
    labelKey="name"
    valueKey="id"
    optionRenderer={optionRenderer}
    valueRenderer={optionRenderer}
    onChange={(i) => onChange(i ? i.id : null)}
    clearable={false}
    id={id}
/>;

const mapStateToProps = (state, ownProps) => ({
    items: availableTags(
        [state.artifacts.items, state.fireteams.items],
        state.tags.items, ownProps.for)
});

export default connect(mapStateToProps)(TagDropDown);
