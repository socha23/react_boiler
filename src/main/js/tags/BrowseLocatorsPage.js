import React from 'react'
import {PropTypes} from 'prop-types'
import {crudList, crudActions} from '../common/crud/crudContainers'
import {LocatorsList} from './LocatorsList'
import {Panel, PanelWithTitle} from '../common/components/Panel'

const BrowseLocatorsPage = ({items, filter, onFilterChange, selected, onSelect}) =>
    <div className="container-fluid">
        <div className="row">
            <div className="col-sm-5 colWithSmallerGutter">
                <Panel>
                    <LocatorsList selected={selected} items={items} onSelect={onSelect}/>
                </Panel>
            </div>
        </div>
    </div>;


class BrowseLocatorsPageContainer extends React.Component {
    state = {
        filter: {}
    };

    onFilterChange = (filter) => {
        this.setState({filter: filter});
    };

    onSelectLocator = (item) => {
        this.context.router.history.push("/tags/locators/" + item.id);
    };

    filterItems = (items) => {
        let currentFilter = this.state.filter;
        return items.filter(i => {
            return true;
        });
    };

    render() {
        return <BrowseLocatorsPage
            {...this.props}
            items={this.filterItems(this.props.items)}
            selected={this.props.items.find(i => i.id == this.props.match.params.id)}
            onSelect={this.onSelectLocator}
            filter={this.state.filter}
            onFilterChange={this.onFilterChange}
        />
    }
}

BrowseLocatorsPageContainer.contextTypes = {
    router: PropTypes.shape({
        history: PropTypes.object.isRequired
    })
};

export default crudList("locators", BrowseLocatorsPageContainer)