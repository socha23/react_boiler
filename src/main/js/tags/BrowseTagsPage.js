import React from 'react'
import {PropTypes} from 'prop-types'
import {crudList, crudActions} from '../common/crud/crudContainers'
import {TagsList} from './TagsList'
import {Panel, PanelWithTitle} from '../common/components/Panel'

const BrowseTagsPage = ({items, filter, onFilterChange, selected, onSelectItem}) =>
    <div className="container-fluid">
        <div className="row">
            <div className="col-sm-5 colWithSmallerGutter">
                <Panel>
                    <TagsList selected={selected} items={items} onSelectItem={onSelectItem}/>
                </Panel>
            </div>
        </div>
    </div>;


class BrowseTagsPageContainer extends React.Component {
    state = {
        filter: {}
    };

    onFilterChange = (filter) => {
        this.setState({filter: filter});
    };

    onSelectTag = (item) => {
        this.context.router.history.push("/tags/" + item.id);
    };

    filterItems = (items) => {
        let currentFilter = this.state.filter;
        return items.filter(i => {
            return true;
        });
    };

    render() {
        return <BrowseTagsPage
            {...this.props}
            items={this.filterItems(this.props.items)}
            selected={this.props.items.find(i => i.id == this.props.match.params.id)}
            onSelect={this.onSelectTag}
            filter={this.state.filter}
            onFilterChange={this.onFilterChange}
        />
    }
}

BrowseTagsPageContainer.contextTypes = {
    router: PropTypes.shape({
        history: PropTypes.object.isRequired
    })
};

export default crudList("tags", BrowseTagsPageContainer)