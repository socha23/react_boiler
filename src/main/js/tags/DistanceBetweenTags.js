import React from 'react'

import {isInside} from './tagHelpers'

export function distance(from, to) {
    let dx = from.position.x - to.position.x;
    let dy = from.position.y - to.position.y;
    return Math.ceil(Math.sqrt(dx * dx + dy * dy));
}

const DistanceBetweenTags = ({from, to}) => {
    if (!from || !to) {
        return <span/>
    } else if (!isInside(from) || !isInside(to)) {
        return <span>?</span>
    } else {
        return <span>{distance(from, to)}m</span>
    }
};

export default DistanceBetweenTags;
