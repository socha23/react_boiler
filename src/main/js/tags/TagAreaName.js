import React from 'react'
import {connect} from 'react-redux'
import {isMissing} from './tagHelpers'
import {getLocatorsByTagId} from './selectors'

export const TagAreaNameComponent = ({tag, style, locatorsByTagId}) => {
    if (!tag) {
        return <span/>
    }
    let color = "#AAA";
    let text = tag.areaName;

    if (isMissing(tag)) {
        color = "#a94442"; // dark red from bootstrap
        text = "poza budynkiem";
    } else if (locatorsByTagId[tag.id]) {
        color = "#a94442"; // dark red from bootstrap
        text = locatorsByTagId[tag.id].name;
    }
    return <small style={{color: color, ...style}}>
        {text}
    </small>
};

const mapStateToProps = (state, ownProps) => {
    let tag = null;

    if (ownProps.tag) {
        tag = ownProps.tag
    } else if (ownProps.tagId) {
        tag = (state.tags.itemsById[ownProps.tagId])
    }
    return {
        tag: tag,
        locatorsByTagId: getLocatorsByTagId(state)
    }
};

export
    default
        connect(mapStateToProps)(TagAreaNameComponent);
