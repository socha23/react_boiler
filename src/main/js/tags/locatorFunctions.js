import store from '../store'

let LOCATORS_BY_ID = null;


store.subscribe(() => {
    LOCATORS_BY_ID = {};
    store.getState().locators.items.forEach(i => LOCATORS_BY_ID[i.id] = i);
});

function getLocatorById(id) {
    if (!LOCATORS_BY_ID || !id) {
        return null;
    }
    return LOCATORS_BY_ID[id];
}

function artifactLocator(artifact) {
    return getLocatorById(artifact.currentLocatorId);
}

module.exports = {
    artifactLocator: artifactLocator,
    getLocatorById: getLocatorById
};