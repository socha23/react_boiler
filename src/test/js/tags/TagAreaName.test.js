import React from 'react'
import {testSnapshot, expectElem} from '../testUtils'

import {TagAreaNameComponent} from 'tags/TagAreaName'

const TAG_INSIDE = {state: "INSIDE", areaName: "inside"};
const TAG_IN_CONTAINER = {state: "IN_CONTAINER", areaName: "in container"};
const TAG_MISSING = {state: "MISSING", areaName: "missing"};

it('renders correctly when inside', testSnapshot(
    <TagAreaNameComponent tag={TAG_INSIDE}/>
));


it('contains area name when inside', () => {
    expectElem(
        <TagAreaNameComponent tag={TAG_INSIDE}/>
    ).toIncludeText("inside")
});

it('contains "poza budynkiem" when missing', () => {
    expectElem(
        <TagAreaNameComponent tag={TAG_MISSING}/>
    ).toIncludeText("poza budynkiem")
});


it('contains "w kontenerze" when in container', () => {
    expectElem(
        <TagAreaNameComponent tag={TAG_IN_CONTAINER}/>
    ).toIncludeText("w kontenerze")
});
