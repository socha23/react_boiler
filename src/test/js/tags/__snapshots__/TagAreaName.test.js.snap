// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when inside 1`] = `
<TagAreaNameComponent
  tag={
    Object {
      "areaName": "inside",
      "state": "INSIDE",
    }
  }
>
  <small
    style={
      Object {
        "color": "#AAA",
      }
    }
  >
    inside
  </small>
</TagAreaNameComponent>
`;
